File: app/src/main/java/com/systemallica/valenbisi/clustering/IconRenderer.java
package com.systemallica.valenbisi.clustering;

import android.content.Context;

import com.google.android.gms.maps.GoogleMap;
import com.google.android.gms.maps.model.MarkerOptions;
import com.google.maps.android.clustering.ClusterManager;
import com.google.maps.android.clustering.view.DefaultClusterRenderer;


public class IconRenderer extends DefaultClusterRenderer<ClusterPoint> {

    private final Context mContext;

    public IconRenderer(Context context, GoogleMap map,
                        ClusterManager<ClusterPoint> clusterManager) {
        super(context, map, clusterManager);

        mContext = context;
    }

    @Override
    protected void onBeforeClusterItemRendered(ClusterPoint item, MarkerOptions markerOptions) {
        markerOptions.icon(item.getIcon());
        markerOptions.snippet(item.getSnippet());
        markerOptions.title(item.getTitle());
        markerOptions.alpha(item.getAlpha());
        markerOptions.visible(item.getVisibility());
        super.onBeforeClusterItemRendered(item, markerOptions);
    }

}

File: app/src/main/java/com/systemallica/valenbisi/clustering/IconRenderer.kt
package com.systemallica.valenbisi.clustering

import android.content.Context

import com.google.android.gms.maps.GoogleMap
import com.google.android.gms.maps.model.MarkerOptions
import com.google.maps.android.clustering.ClusterManager
import com.google.maps.android.clustering.view.DefaultClusterRenderer


class IconRenderer(
    context: Context, map: GoogleMap,
    clusterManager: ClusterManager<ClusterPoint>
) : DefaultClusterRenderer<ClusterPoint>(context, map, clusterManager) {

    override fun onBeforeClusterItemRendered(item: ClusterPoint?, markerOptions: MarkerOptions?) {
        markerOptions!!.icon(item!!.icon)
        markerOptions.snippet(item.snippet)
        markerOptions.title(item.title)
        markerOptions.alpha(item.alpha!!)
        markerOptions.visible(item.visibility!!)
        super.onBeforeClusterItemRendered(item, markerOptions)
    }
}